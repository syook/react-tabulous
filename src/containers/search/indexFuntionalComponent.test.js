import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import SearchProvider from './indexFuntionalComponent';
// import { it } from 'date-fns/locale';

configure({ adapter: new Adapter() });

const rawData = [
  { id: 1, description: 'Dev', isDeleted: false, is_completed: true, name: 'Harsh Singh' },
  { id: 2, description: 'QA', isDeleted: false, name: 'Prakash' },
];
const searchKeys = {
  Description: true,
  Name: true,
};
const tableData = [
  { id: 1, objIndex: 0, name: 'Harsh Singh', description: 'Dev' },
  { id: 2, objIndex: 1, name: 'Prakash', description: 'QA' },
];

describe('SearchProvider', () => {
  it('renders without crashing', () => {
    shallow(<SearchProvider />);
  });

  it('accepts props', () => {
    const wrapper = mount(<SearchProvider rawData={rawData} searchKeys={searchKeys} tableData={tableData} />);
    expect(wrapper.props().rawData).toEqual(rawData);
    expect(wrapper.props().searchKeys).toEqual(searchKeys);
    expect(wrapper.props().tableData).toEqual(tableData);
  });

  it('matches snapshot', () => {
    const tree = shallow(<SearchProvider />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render search input', () => {
    const wrapper = shallow(<SearchProvider />);
    expect(wrapper.find('SearchComponent').exists()).toBe(true);
  });

  it('initial disabled prop should be true', () => {
    const wrapper = shallow(<SearchProvider />);
    expect(wrapper.find('SearchComponent').prop('disabled')).toBe(true);
  });

  it('name prop in SearchComponent will be undefined if tableName is not passed', () => {
    const wrapper = shallow(<SearchProvider />);
    expect(wrapper.find('SearchComponent').prop('name')).toBe(undefined);
  });

  it('name prop in SearchComponent should be passed tableName prop', () => {
    const tableName = 'data table';
    const wrapper = shallow(<SearchProvider tableName={tableName} />);
    expect(wrapper.find('SearchComponent').prop('name')).toBe(tableName);
  });

  it('should matches snapshot if data is not empty', () => {
    const tree = shallow(<SearchProvider tableData={tableData} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should match snapshot if tableData is present', () => {
    const wrapper = shallow(<SearchProvider tableData={tableData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshop if tableData is not empty', () => {
    const wrapper = mount(<SearchProvider tableData={tableData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshop if tableData is empty', () => {
    const wrapper = mount(<SearchProvider tableData={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contain text "No data to Display" if tableData is empty', () => {
    const wrapper = mount(<SearchProvider tableData={[]} />);
    expect(wrapper.find('SearchProvider').contains('No data to Display')).toBe(true);
  });

  it('should contain text "No TableOne to Display" if tableData is empty and tableName is passed', () => {
    const wrapper = mount(<SearchProvider tableData={[]} tableName={'TableOne'} />);
    expect(wrapper.find('SearchProvider').contains('No TableOne to Display')).toBe(true);
  });
});
