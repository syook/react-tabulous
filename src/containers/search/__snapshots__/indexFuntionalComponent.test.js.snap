// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProvider matches snapshot 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "columnName": null,
        "columnType": null,
        "count": undefined,
        "data": Array [],
        "direction": null,
        "rawData": undefined,
        "rowsPerPage": 10,
        "rowsPerPageFromSearch": 10,
        "searchText": "",
        "updateRowsPerPage": [Function],
        "updateRowsSortParams": [Function],
      }
    }
  >
    <SearchComponent
      disabled={true}
      onChangeSearchText={[Function]}
    />
    <div
      style={
        Object {
          "padding": "0 15px",
        }
      }
    >
      <div
        className="noRecordsDiv"
      >
        No data to Display
      </div>
    </div>
  </ContextProvider>
</div>
`;

exports[`SearchProvider should match snapshop if tableData is empty 1`] = `
<SearchProvider
  tableData={Array []}
>
  <div>
    <SearchComponent
      disabled={true}
      onChangeSearchText={[Function]}
    >
      <div>
        <div
          style={
            Object {
              "position": "relative",
              "right": "0px",
              "top": "-30px",
            }
          }
        >
          <Input
            disabled={true}
            iconPosition="left"
            onChange={[Function]}
            placeholder="Search..."
            style={
              Object {
                "borderRadius": "3px",
                "color": "#667797",
                "marginLeft": "10px",
                "maxWidth": "300px",
                "position": "absolute",
                "right": "15px",
                "top": "35px",
                "zIndex": 9,
              }
            }
            type="text"
          >
            <div
              className="ui disabled left icon input"
              style={
                Object {
                  "borderRadius": "3px",
                  "color": "#667797",
                  "marginLeft": "10px",
                  "maxWidth": "300px",
                  "position": "absolute",
                  "right": "15px",
                  "top": "35px",
                  "zIndex": 9,
                }
              }
            >
              <Icon
                as="i"
                key=".0"
                name="search"
              >
                <i
                  aria-hidden="true"
                  className="search icon"
                  onClick={[Function]}
                />
              </Icon>
              <input
                className="tabulous-searchInput"
                disabled={true}
                key=".1"
                onChange={[Function]}
                placeholder="Search..."
                tabIndex={-1}
                type="text"
                value=""
              />
              <Icon
                as="i"
                disabled={true}
                key=".2"
                name="close"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "left": "unset",
                    "pointerEvents": "auto",
                    "position": "absolute",
                    "right": "0px",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="close disabled icon"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "left": "unset",
                      "pointerEvents": "auto",
                      "position": "absolute",
                      "right": "0px",
                    }
                  }
                />
              </Icon>
            </div>
          </Input>
        </div>
      </div>
    </SearchComponent>
    <div
      style={
        Object {
          "padding": "0 15px",
        }
      }
    >
      <div
        className="noRecordsDiv"
      >
        No data to Display
      </div>
    </div>
  </div>
</SearchProvider>
`;

exports[`SearchProvider should match snapshop if tableData is not empty 1`] = `
<SearchProvider
  tableData={
    Array [
      Object {
        "description": "Dev",
        "id": 1,
        "name": "Harsh Singh",
        "objIndex": 0,
      },
      Object {
        "description": "QA",
        "id": 2,
        "name": "Prakash",
        "objIndex": 1,
      },
    ]
  }
>
  <div>
    <SearchComponent
      disabled={false}
      onChangeSearchText={[Function]}
    >
      <div>
        <div
          style={
            Object {
              "position": "relative",
              "right": "0px",
              "top": "-30px",
            }
          }
        >
          <Input
            disabled={false}
            iconPosition="left"
            onChange={[Function]}
            placeholder="Search..."
            style={
              Object {
                "borderRadius": "3px",
                "color": "#667797",
                "marginLeft": "10px",
                "maxWidth": "300px",
                "position": "absolute",
                "right": "15px",
                "top": "35px",
                "zIndex": 9,
              }
            }
            type="text"
          >
            <div
              className="ui left icon input"
              style={
                Object {
                  "borderRadius": "3px",
                  "color": "#667797",
                  "marginLeft": "10px",
                  "maxWidth": "300px",
                  "position": "absolute",
                  "right": "15px",
                  "top": "35px",
                  "zIndex": 9,
                }
              }
            >
              <Icon
                as="i"
                key=".0"
                name="search"
              >
                <i
                  aria-hidden="true"
                  className="search icon"
                  onClick={[Function]}
                />
              </Icon>
              <input
                className="tabulous-searchInput"
                disabled={false}
                key=".1"
                onChange={[Function]}
                placeholder="Search..."
                type="text"
                value=""
              />
              <Icon
                as="i"
                disabled={false}
                key=".2"
                name="close"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "left": "unset",
                    "pointerEvents": "auto",
                    "position": "absolute",
                    "right": "0px",
                  }
                }
              >
                <i
                  aria-hidden="true"
                  className="close icon"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "left": "unset",
                      "pointerEvents": "auto",
                      "position": "absolute",
                      "right": "0px",
                    }
                  }
                />
              </Icon>
            </div>
          </Input>
        </div>
      </div>
    </SearchComponent>
  </div>
</SearchProvider>
`;

exports[`SearchProvider should match snapshot if tableData is present 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "columnName": null,
        "columnType": null,
        "count": undefined,
        "data": Array [],
        "direction": null,
        "rawData": undefined,
        "rowsPerPage": 10,
        "rowsPerPageFromSearch": 10,
        "searchText": "",
        "updateRowsPerPage": [Function],
        "updateRowsSortParams": [Function],
      }
    }
  >
    <SearchComponent
      disabled={false}
      onChangeSearchText={[Function]}
    />
    <div
      style={
        Object {
          "padding": "0 15px",
        }
      }
    >
      <div
        className="noRecordsDiv"
      >
        No Results Found
      </div>
    </div>
  </ContextProvider>
</div>
`;

exports[`SearchProvider should matches snapshot if data is not empty 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "columnName": null,
        "columnType": null,
        "count": undefined,
        "data": Array [],
        "direction": null,
        "rawData": undefined,
        "rowsPerPage": 10,
        "rowsPerPageFromSearch": 10,
        "searchText": "",
        "updateRowsPerPage": [Function],
        "updateRowsSortParams": [Function],
      }
    }
  >
    <SearchComponent
      disabled={false}
      onChangeSearchText={[Function]}
    />
    <div
      style={
        Object {
          "padding": "0 15px",
        }
      }
    >
      <div
        className="noRecordsDiv"
      >
        No Results Found
      </div>
    </div>
  </ContextProvider>
</div>
`;
